### PR TITLE
fix: remove dup apply pass on the same physical op

### LINKED ISSUE #437: Bug: fail to compile and run some SQL statements with OpenMLDB-batch when enable batch_window_parallelization
Issue tracker is **ONLY** used for reporting bugs. New features should be discussed on our discussion

We fail to compile and run SQL with OpenMLDB-batch when enable `batch_window_parallelization`
```sql
-- t1: ["col0 string", "col1 int", "col2 int"]
-- t2: ["str0 string", "str1 string", "col0 int", "col1 int"]
SELECT sum(t1.col1) over w1 as sum_t1_col1, t2.str1 as t2_str1 FROM t1
               last join t2 order by t2.col1
               on t1.col1 = t2.col1 and t1.col2 = t2.col0
               WINDOW w1 AS (PARTITION BY t1.col2 ORDER BY t1.col1
               ROWS_RANGE BETWEEN 3 PRECEDING AND CURRENT ROW);
```

## Expected Behavior

Compile and run successly for SQL

## Current Behavior

Fail to compile SQL.
```
Fail to find column id #48 in current schema context
    (At /Users/chenjing/work/chenjing/OpenMLDB/hybridse/src/vm/sql_compiler.cc:260)
    (At /Users/chenjing/work/chenjing/OpenMLDB/hybridse/src/vm/sql_compiler.cc:166)
    (Caused by) Fail to generate physical plan batch mode
    (At /Users/chenjing/work/chenjing/OpenMLDB/hybridse/src/vm/transform.cc:1489)
    (Caused by) Fail to generate functions for physical plan: LIMIT(limit=10)
  SIMPLE_PROJECT(sources=(sum_t1_col1, t2_str1))
    JOIN(type=kJoinTypeConcat)
      SIMPLE_PROJECT(sources=(t2.str1 -> t2_str1))
        JOIN(type=LastJoin, right_sort=(t2.col1 ASC), condition=, left_keys=(t1.col1,t1.col2), right_keys=(t2.col1,t2.col0), index_keys=)
          DATA_PROVIDER(table=t1)
          DATA_PROVIDER(table=t2)
      PROJECT(type=WindowAggregation)
        +-WINDOW(partition_keys=(t1.col2), orders=(t1.col1 ASC), range=(t1.col1, -3, 0))
        JOIN(type=LastJoin, right_sort=(t2.col1 ASC), condition=, left_keys=(t1.col1,t1.col2), right_keys=(t2.col1,t2.col0), index_keys=)
          DATA_PROVIDER(table=t1)
          DATA_PROVIDER(table=t2)
    (At /Users/chenjing/work/chenjing/OpenMLDB/hybridse/src/vm/transform.cc:197)
    (At /Users/chenjing/work/chenjing/OpenMLDB/hybridse/src/vm/transform.cc:197)
    (At /Users/chenjing/work/chenjing/OpenMLDB/hybridse/src/vm/transform.cc:197)
    (At /Users/chenjing/work/chenjing/OpenMLDB/hybridse/src/vm/transform.cc:290)
    (Caused by) Instantiate 0th native function "__internal_sql_codegen_12" failed at node:
SIMPLE_PROJECT(sources=(t2.str1 -> t2_str1))
  JOIN(type=LastJoin, right_sort=(t2.col1 ASC), condition=, left_keys=(t1.col1,t1.col2), right_keys=(t2.col1,t2.col0), index_keys=)
    DATA_PROVIDER(table=t1)
    DATA_PROVIDER(table=t2)
    (At /Users/chenjing/work/chenjing/OpenMLDB/hybridse/src/codegen/fn_let_ir_builder.cc:130)
    (Caused by) Build expr failed at 0:
+-expr[get field]
  +-input:
    +-expr[id]
      +-var: %44(row)
  +-column_id: 48
  +-column_name: str1
    (At /Users/chenjing/work/chenjing/OpenMLDB/hybridse/src/codegen/fn_let_ir_builder.cc:215)
    (Caused by) Fail to codegen project expression: #48:str1
    (At /Users/chenjing/work/chenjing/OpenMLDB/hybridse/src/codegen/expr_ir_builder.cc:162)
    (At /Users/chenjing/work/chenjing/OpenMLDB/hybridse/src/codegen/expr_ir_builder.cc:835)
    (Caused by) Fail to resolve column #48:str1 from row
    (At /Users/chenjing/work/chenjing/OpenMLDB/hybridse/src/vm/schemas_context.cc:275)
    (Caused by) Fail to find column id #48 in current schema context(#1: 0, 0),(#2: 0, 1),(#3: 0, 2),(#51: 1, 0),(#52: 1, 1),(#53: 1, 2),(#54: 1, 3),
column_name_map_:col0, col1, col2, str0, str1, 
```

## Possible Solution

I think the problem is raised by the incorrect column resolve and inappropriate schema update.

In fact, things change after we applying for optimization passes on the physical plan. So there must be something wrong with the optimization passes. 
We also observe that the `SIMPLE_PROJECT(sources=(t2.str1 -> t2_str1))` and `PROJECT(type=WindowAggregation)` share the same `JOIN` node.  If we process passes in a **post-DFS** order,  we might update `JOIN` node firstly and update `SimpleProject` later  (with on new `JOIN` schema), and `JOIN` again and update` WindowAggregation Project` finally. 
So, based on the observation above, I strongly concern that we might apply optimization on `JOIN` twice so that SIMPLE_PROJCT can't be up-to-date.  That's why we fail to resolve column `t2_str1` on `SimpleProject`.

#### Solution
1. Keep track of if a `PhysicalNode` has been visited or not.
2. Stop process optimization if a node has been visited before.

## Steps to Reproduce

1.
2.
3.
4.

### PR DESCRIPTION
**Pull request by jingchen2222**
* **Please check if the PR fulfills these requirements**

- The commit message follows our guidelines
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix.

* **What is the current behavior?** (You can also link to an open issue here)
#437

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: hybridse/java/hybridse-native/pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	<parent>
         <artifactId>hybridse-java</artifactId>
         <groupId>com.4paradigm.hybridse</groupId>
-        <version>0.2.4-SNAPSHOT</version>
+        <version>0.2.4.20210928-SNAPSHOT</version>
```
Comment (aceforeverd) [inline hybridse/java/hybridse-native/pom.xml]: This change should reverted before merge into main

Comment (aceforeverd) [reply hybridse/java/hybridse-native/pom.xml]: Could we just use single version in https://s01.oss.sonatype.org/content/repositories/snapshots/com/4paradigm/hybridse/hybridse-sdk/0.2.4.20210928-SNAPSHOT/ ? 
E.g 'hybridse-sdk-0.2.4.20210928-20210928.183823-1.jar'

**Review by zekai427 — APPROVED** [review]

File: hybridse/java/prepare_release.sh [hunk 2]
```diff
@@ -32,21 +32,36 @@ fi
...
-# rm semVer number from VERSION
+# rm semVer number from VERSION of 3 numbers MAJOR.MINOR.PATCH
 #  0.1.2 -> ''
 #  0.1.2-SNAPSHOT -> '-SNAPSHOT'
 #  0.1.2.beta1    -> '.beta1'
+#  0.2.1.0928     -> '.0928'
 # shellcheck disable=SC2001
 SUFFIX_VERSION=$(echo "$VERSION" | sed -e 's/^[0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*//')
+
+DEBUG_SUFFIX_VERSION=$(echo "$SUFFIX_VERSION" | sed -e 's/\.[0-9][0-9]*//')
```
Comment (aceforeverd) [inline hybridse/java/prepare_release.sh:43]: add `# shellcheck disable=SC2001` in above can ignore the error

File: hybridse/tools/hybridse_build_jsdk.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# limitations under the License.
+
+set -eE
+
+# goto toplevel directory
+cd "$(dirname "$0")/.."
+
+# install thirdparty hybrise
+HYBRIDSE_THIRDPARTY="$(pwd)/thirdparty"
+../steps/setup_thirdparty.sh ${HYBRIDSE_THIRDPARTY}
```
Comment (aceforeverd) [inline hybridse/tools/hybridse_build_jsdk.sh]: double quote `${HYBRIDSE_THIRDPARTY}`

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Pull request merged**